### PR TITLE
Feature/aa 1006 update net6

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#1006

### CODE REVIEW

File: Api/Services/Accommodations/Bookings/BookingExtensions.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace HappyTravel.Edo.Api.Services.Accommodations.Bookings
     public static class BookingExtensions
     {
         public static DateTime GetPayDueDate(this Data.Bookings.Booking booking)
-            => (booking.DeadlineDate == null || booking.DeadlineDate == DateTime.MinValue ? booking.CheckInDate : booking.DeadlineDate.Value);
+            => (booking.DeadlineDate?.DateTime == null || booking.DeadlineDate?.DateTime == DateTime.MinValue ? booking.CheckInDate.DateTime : booking.DeadlineDate.Value.DateTime);
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/BookingExtensions.cs:9]: May not get datetime when checking for null

File: Api/AdministratorServices/AdminAgencyManagementService.cs [hunk 2]
```diff
@@ -53,7 +53,16 @@ public async Task<Result<AgencyInfo>> Get(int agencyId, string languageCode = Lo
                     join ra in _context.Agencies on a.Ancestors.Any() ? a.Ancestors[0] : a.Id equals ra.Id
                     from markupFormula in _context.DisplayMarkupFormulas.Where(f => f.AgencyId == a.Id && f.AgentId == null).DefaultIfEmpty() 
                     where a.Id == agencyId
-                    select a.ToAgencyInfo(a.ContractKind, ra.VerificationState, ra.Verified, c.Names, languageCode, markupFormula == null ? string.Empty : markupFormula.DisplayFormula))
+                    select a.ToAgencyInfo(a.ContractKind,
```
Comment (mingazhev) [inline Api/AdministratorServices/AdminAgencyManagementService.cs:56]: Please add argument names as well

File: HappyTravel.Edo.Data/Bookings/CancellationPolicy.cs [hunk 3]
```diff
@@ -9,13 +9,13 @@ public class CancellationPolicy
...
         [JsonConstructor]
-        public CancellationPolicy(DateTime fromDate, double percentage)
+        public CancellationPolicy(DateTimeOffset fromDate, double percentage)
         {
             FromDate = fromDate;
             Percentage = percentage;
         }
         
-        public DateTime FromDate { get; set; }
+        public DateTimeOffset FromDate { get; set; }
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Bookings/CancellationPolicy.cs:18]: No need to change there as I understand. This is a JSON column

File: HappyTravel.Edo.Data/Bookings/Deadline.cs [hunk 4]
```diff
@@ -10,15 +10,15 @@ public class Deadline
...
+        public Deadline(DateTimeOffset? date, List<CancellationPolicy> policies, List<string> remarks, bool isFinal)
         {
             Date = date;
             IsFinal = isFinal;
             Policies = policies ?? new List<CancellationPolicy>();
             Remarks = remarks ?? new List<string>(0);
         }
         
-        public DateTime? Date { get; set; }
+        public DateTimeOffset? Date { get; set; }
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Bookings/Deadline.cs:21]: No need to change there, this is a JSON field

File: Api/Services/Agents/AgencyService.cs [hunk 5]
```diff
@@ -163,7 +163,16 @@ private async Task<Result<AgencyInfo>> GetAgencyInfo(int agencyId, string langua
                     join ra in _context.Agencies on a.Ancestors.Any() ? a.Ancestors[0] : a.Id equals ra.Id
                     from markupFormula in _context.DisplayMarkupFormulas.Where(f => f.AgencyId == a.Id && f.AgentId == null).DefaultIfEmpty() 
                     where a.Id == agencyId
-                    select a.ToAgencyInfo(a.ContractKind, ra.VerificationState, ra.Verified, c.Names, languageCode, markupFormula == null ? string.Empty : markupFormula.DisplayFormula))
+                    select a.ToAgencyInfo(a.ContractKind, 
+                        ra.VerificationState,
```
Comment (mingazhev) [inline Api/Services/Agents/AgencyService.cs:167]: Please add argument name as soon as you changed the formatting

File: Api/Services/Accommodations/Bookings/BookingExtensions.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/BookingExtensions.cs]: Fixed

File: Api/Services/Agents/AgencyService.cs (hunk 5 above)
Comment (DFSko) [reply Api/Services/Agents/AgencyService.cs:167]: CS0853: An expression tree may not contain a named argument specification

File: Api/AdministratorServices/AdminAgencyManagementService.cs (hunk 2 above)
Comment (DFSko) [reply Api/AdministratorServices/AdminAgencyManagementService.cs:56]: CS0853: An expression tree may not contain a named argument specification

File: HappyTravel.Edo.Data/Bookings/CancellationPolicy.cs (hunk 3 above)
Comment (DFSko) [reply HappyTravel.Edo.Data/Bookings/CancellationPolicy.cs]: Fixed

File: HappyTravel.Edo.Data/Bookings/Deadline.cs (hunk 4 above)
Comment (DFSko) [reply HappyTravel.Edo.Data/Bookings/Deadline.cs]: Fixed

**Pull request merged**